docs/proposals/20220712-cluster-api-addon-orchestration.md

```diff
@@ -0,0 +1,450 @@
+
+This is a link to the prototype for [ClusterAddonProvider for Helm](https://github.com/Jont828/cluster-api-addon-provider-helm), and a demo can be found in this [Cluster API Office Hours recording](https://www.youtube.com/watch?v=dipR1Tzh4jE). The prototype implements HelmChartProxy and HelmReleaseProxy as well as controllers for both CRDs.
+
+#### HelmChartProxy
+
+HelmChartProxy is a namespaced CRD that serves as the user interface for defining and configuring Helm charts as well as selecting which workload clusters it will be installed on. 
+
+```go
+// HelmChartProxySpec defines the desired state of HelmChartProxy.
+type HelmChartProxySpec struct {
+  // ClusterSelector selects Clusters in the same namespace with a label that matches the specified label selector. The Helm 
+  // chart will be installed on all selected Clusters. If a Cluster is no longer selected, the Helm release will be uninstalled.
+  ClusterSelector metav1.LabelSelector `json:"clusterSelector"`
+
+  // ChartName is the name of the Helm chart in the repository.
+  ChartName string `json:"chartName"`
+
+  // RepoURL is the URL of the Helm chart repository.
+  RepoURL string `json:"repoURL"`
+
+  // ReleaseName is the release name of the installed Helm chart. If it is not specified, a
+  // name will be generated.
+  // +optional
+  ReleaseName string `json:"releaseName,omitempty"`
+
+  // ReleaseNamespace is the namespace the Helm release will be installed on each selected
+  // Cluster. If it is not specified, the default namespace.
+  // +optional
+  ReleaseNamespace string `json:"namespace,omitempty"`
+
+  // Version is the version of the Helm chart. If it is not specified, the chart will use 
+  // and be kept up to date with the latest version.
+  // +optional
+  Version string `json:"version,omitempty"`
+
+  // ValuesTemplate is an inline YAML representing the values for the Helm chart. This YAML supports Go templating to reference
+  // fields from each selected workload Cluster and programatically create and set values.
+  // +optional
+  ValuesTemplate string `json:"valuesTemplate,omitempty"`
+}
+
+// HelmChartProxyStatus defines the observed state of HelmChartProxy.
+type HelmChartProxyStatus struct {
+  // Conditions defines current state of the HelmChartProxy.
+  // +optional
+  Conditions clusterv1.Conditions `json:"conditions,omitempty"`
+
+  // MatchingClusters is the list of references to Clusters selected by the ClusterSelector.
+  // +optional
+  MatchingClusters []corev1.ObjectReference `json:"matchingClusters"`
+}
+```
```

Comment:
does this have a status?

Reply:
Do you mean the `MatchingClusters` field or the HelmChartProxyStatus?